--- lower_tensors.cc.orig	2025-01-01 00:00:00.000000000 +0000
+++ lower_tensors.cc	2025-01-01 00:00:01.000000000 +0000
@@ -823,7 +823,7 @@
 
   Value llvm_value =
       b.create<UnrealizedConversionCastOp>(llvm_input_ty, value).getResult(0);
   Value alloca = entry_builder.create<ml::AllocaOp>(
-      ptr_ty, llvm_input_ty, b.create<ml::ConstantOp>(b.getI32Type(), 1));
+      ptr_ty, llvm_input_ty, b.create<ml::ConstantOp>(b.getI32Type(), 1), /*address_space=*/5);
   b.create<ml::StoreOp>(llvm_value, alloca);
   auto result = b.create<ml::LoadOp>(llvm_result_ty, alloca).getResult();
   return b.create<UnrealizedConversionCastOp>(ty, result).getResult(0);
